### ISSUE TITLE
Support async/await.

### ISSUE THREAD

**Issue by anhlq-supp** [apollographql/apollo-ios#3091]
### Use case

Can you support async/await soon in the next version?

### Describe the solution you'd like

Need another interface that support async/await method in Swift
Something like
```
fetch<Query: GraphQLQuery>(query: Query,
 cachePolicy: CachePolicy = .default,
contextIdentifier: UUID? = nil,
queue: DispatchQueue = .main) async ->  Result<Query.Data?, Error>
                                                            ```

**Comment by jimisaacs** [comment]
The issue is closed, but there's good passed and actually recent discussion there https://github.com/apollographql/apollo-ios/issues/2216

**Comment by calvincestari** [comment]
Thanks for linking to that issue @jimisaacs, there's been some great discussion in there this week.

@anhlq-supp I'm going to close this issue as there isn't anything immediately actionable for our team. Our public [roadmap](https://github.com/apollographql/apollo-ios/blob/main/ROADMAP.md) details the items we're focused on now and in the short-term ahead. Supporting the new Swift Concurrency features is definitely included in the 2.0 work.
